server.js

```diff
@@ -4,20 +4,84 @@ const cors = require("cors");
 const app = express();
 
 app.use(cors());
+app.use(express.json());
 
 const welcomeMessage = {
   id: 0,
   from: "Bart",
   text: "Welcome to CYF chat system!",
 };
 
-//This array is our "data store".
-//We will start with one message in the array.
-//Note: messages will be lost when Glitch restarts our server.
 const messages = [welcomeMessage];
 
 app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
-app.listen(process.env.PORT);
+app.post("/messages", (request, response) => {
+  if (!request.body.from || !request.body.text) {
+    response.sendStatus(400);
+    return;
+  }
+  const createdMessage = {
+    id: Number(new Date()),
+    from: request.body.from,
+    text: request.body.text,
+  };
+
+  messages.push(createdMessage);
+  console.log(createdMessage);
+  response.status(201).json(createdMessage);
+});
+
+app.get("/messages", (request, response) => {
+  response.json(messages);
+});
+
+app.get("/messages/search", (request, response) => {
+  let foundMessages = messages;
+  if (request.query.text) {
+    foundMessages = foundMessages.filter(
+      (message) =>
+        message.text.toLowerCase().indexOf(request.query.text.toLowerCase()) >
+        -1
+    );
+  }
+  if (foundMessages) {
+    response.json(foundMessages);
+  }
```

Comment:
You don't send a response if no messages are found - this will cause the request to hang

Reply:
Thanks James. Fixed it.